app/helpers/payment_details_helper.rb

```diff
@@ -19,33 +22,36 @@ def link_to_payment_details(text, options = {})
   end
 
   def payment_details_path(merchant_account = site_account, url_params = {})
-    return "" if ["bogus" ,""].include?(merchant_account.payment_gateway_type.to_s)
-    named_route = [:admin, :account, merchant_account.payment_gateway_type]
-    polymorphic_path(named_route, url_params)
+    payment_gateway_type = merchant_account.payment_gateway_type
+    return '' if [:bogus, nil].include?(payment_gateway_type)
```

Comment:
is there a test for this. It just looks a little risky what type is `payment_gateway_type`

Reply:
This was added 11 years ago.. you seriously asking? LOL The whole thing is a bit of a mess so I wouldn't expect none of this is properly tested.
This were the "tests" that were implemented. At least there were a TODO 😂

Reply:
precisely will be very useful to have an example when this `bogus` value may appear. I remember seeing that before and it is unnerving.

Reply:
Right. I think it's only enabled in development env, that's probably why the didn't bother testing it. Anyway, I don't think it is this PR's responsibility **but** I will take a good look into this and add some more tests.